### PR TITLE
tests: Update tests which call config status

### PR DESCRIPTION
**Pull request by dheyay**
Fixes: #2996

## Why is this needed?

This PR solves all of our problems because  a bug was introduced earlier which allowed the unit tests to read a system file, which should not be happening.

## Checklist

 - [x] I have updated or added any unit tests accordingly
 - [ ] I have updated or added any integration tests accordingly
 - [ ] Changes here need to be documented, and this was done in: 

## Does this PR require extra reviews?

 - [ ] Yes
 - [ ] No

### CODE REVIEW

File: uaclient/cli/tests/test_cli_attach.py [hunk 1]
```diff
@@ -553,6 +560,7 @@ def test_attach_when_one_service_fails_to_enable(
         FakeConfig,
         event,
     ):
+        m_public_config.return_value = UserConfigData()
```
Comment (orndorffgrant) [inline uaclient/cli/tests/test_cli_attach.py:563]: `return_value` is already set above in the decorator, so no need to do it again here
```suggestion
```

**Review by orndorffgrant — APPROVED** [review]
Looks good! Just a nit

**Comment by dheyay** [comment]
@orndorffgrant should be good now!

**Review by renanrodrigo — APPROVED** [review]
This is good for now.
@lucasmoura correctly mentioned this should be a fixture, so we don't leak it anywhere in the tests - but that is true for all state files isn't it?

**Pull request merged**